datafusion/src/physical_plan/expressions/average.rs

```diff
@@ -86,11 +94,15 @@ impl Avg {
         // Average is always Float64, but Avg::new() has a data_type
         // parameter to keep a consistent signature with the other
         // Aggregate expressions.
-        assert_eq!(data_type, DataType::Float64);
-
-        Self {
-            name: name.into(),
-            expr,
+        match data_type {
```

Comment:
The comment above looks out of date -- I think it should simply be removed. 

And perhaps we can change this code so it doesn't use `unreachable` as I think it would be fairly easy to reach this code by calling `Avg::new(..)` with some incorrect paramters

How about something like

```rust
assert!(matches!(data_type, DataType::Float64 | DataType::Decimal(_, _)));
```

Which I think might be easier to diagnose if anyone hits it

Reply:
grate comments